File: 118-pascals-triangle/pascals-triangle.py
class Solution(object):
    def generateRow(self,row):
        ans= 1
        temp = []
        temp.append(1)
        for col in range(1,row):
            ans = ans*(row-col)
            ans = ans//col
            temp.append(ans)
        return temp
    def generate(self, numRows):
        """
        :type numRows: int
        :rtype: List[List[int]]
        """
        ans=[]
        for i in range(1,numRows+1):
            ans.append(self.generateRow(i))
        return ans
        

File: 118-pascals-triangle/pascals-triangle.cpp
class Solution {
public:
vector<int>GenerateRow(int row){
    long long ans =1;
    vector<int>tempans;
    tempans.push_back(1);

    for(int i =1;i<row;i++){
        ans = ans*(row-i);
        ans =ans/i;
        tempans.push_back(ans);
    }
    return tempans;
}
    vector<vector<int>> generate(int n) {
        vector<vector<int>>ans;
        for(int i =1;i<=n;i++){
            vector<int>temp = GenerateRow(i);
            ans.push_back(temp);
        }
        return ans;
    }
};